packages/testcontainers/src/generic-container/generic-container-builder.ts

```diff
@@ -105,3 +138,17 @@ export class GenericContainerBuilder {
       .reduce((prev, next) => ({ ...prev, ...next }), {} as RegistryConfig);
   }
 }
+
+async function newDockerignoreFilter(context: string): Promise<(path: string) => boolean> {
```

Comment:
Only to curiosity in your approach, why `newDockerignoreFilter` function  is not part from `class GenericContainerBuilder` as a private method take into account that you are only using it to this class responsabilities?

Reply:
I believe class methods should access or modify the object instance. 

In this case, `newDockerignoreFilter` does not access nor modify any object fields. It can be seen as a factory method. It is a function that, given a context path, returns a `.dockerignore` based filter function.